### ISSUE TITLE
Decription tools

### ISSUE THREAD

**Issue by iz4tow** [lwvmobile/dsd-fme#80]
greetings to everyone........can basic privacy be decrypted with this version of dsd? functions like dsdplus can be implemented ..... such as id display and others? thanks

_Originally posted by @romanremus in https://github.com/lwvmobile/dsd-fme/issues/79#issuecomment-1219093216_

It would be nice to have decryption features

**Comment by romanremus** [comment]
the version that Eric Herve has, I saw that it can decrypt......but changes are needed in dmr_encryption.c

**Comment by lwvmobile** [comment]
> can basic privacy be decrypted with this version of dsd?

Short answer is yes. The code is in there for 'DMRA' or 'Moto' BP. You can enter a key value with -K or use the appropriate entry box in the ncurses menu.

> such as id display and others?
I probably could have the ncurses terminal display the actual name of which encryption type is in use, along with the value of the alg_id. I actually do so for NXDN and P25 on known encryption types, within reason. P25 has a long list of potential ones, but in practice, I've only ever seen a few of them.

> It would be nice to have decryption features
I'm not going to argue with you. I am not opposed to the use of tools as tools. While I don't personally believe the inclusion of decryption methods is of itself bad, I have to consider the possibility for legal issues revolving around the law of different states and countries, as well as from a patent perspective and civil/business. There are a few projects out there now slowly starting to implement some of the lesser decryption methods for systems like RC4 and BP key and they haven't had any issues that I am aware of. I'll continue to monitor those projects and see if anything comes of it. I am also aware of one project in particular that had received a DMCA notice for including some form of decryption. So, again, I'm keeping my eyes and ears open to what's going on with other projects as well.

So, in short, for the foreseeable future, I will not be including any form of advanced or enhanced decryption methods.

**Comment by romanremus** [comment]
perfect .......I sent here the link with the raw file......as for RC4 do you think something can be done ???? congratulations for your code ....however, I can't find where you implemented the keys for basic privacy      https://ufile.io/qonemoqk

**Comment by romanremus** [comment]
it's perfect like that ..... please send us your scripts ... and please tell us in your code where to find the keys for basic privacy ????

**Comment by lwvmobile** [comment]
> and please tell us in your code where to find the keys for basic privacy ????

If you are referring to which code in DSD-FME pertains to the handling of BP, then you should look at either src/dmr_process_voice.c for the LEH audio decoding method, or inside of dsd_mbe.c for the stereo method. I feel like it would be pretty hard to miss if you look at those files.

Again, this is for those who already know their key id AND have legitimate access to that system. I will not support or tell users how to go about figuring out how to reverse a key.

**Comment by lwvmobile** [comment]
> perfect .......I sent here the link with the raw file......as for RC4 do you think something can be done ???? congratulations for your code ....however, I can't find where you implemented the keys for basic privacy https://ufile.io/qonemoqk

Okay, thanks for the sample submission. I gave it a quick analysis, and its an interesting sample, but I'm afraid its not very useful. Sadly, that sample has a lot of errors in it, I suspect bad signal, marginal signal, or lots of noise in it. Its very jittery to say the least.
Do you have any cleaner samples of the same or a similar system?

**Comment by romanremus** [comment]
I saw that you added the code for LRRP ???? Can it decode something like this???? I always get rtl error when I want to compile it with cygwin for windows....
could we also add an event log??? Thank you very much

**Comment by iz4tow** [comment]
Try to use wsl. If you have windows 7 I will provide a docker installation
guide for win 7

**Comment by lwvmobile** [comment]
> I saw that you added the code for LRRP ????

The code is currently able to decode LRRP, but there are some hit or miss samples I have, and also some samples that I don't believe are an actual LRRP or Embedded GPS signalling, but the 3/4 Rate Data seems to convey some form of coordinates. I believe some version of DSDPlus classify this as LOCN.

> could we also add an event log???

These is already, in one fashion, an event log of sorts. The Ncurses Terminal window will show call history and also other things of interest. Also, in Linux and the Cygwin Terminal, a user can use combination of the tail command and a selective grep command to pull items of interest out of the log file in realtime, thus making that an event log monitor of sorts as well.

**Comment by romanremus** [comment]
please forget about the error that appears when I try to compile with cygwin the latest version of dsd with LRRP ....... and I don't know how to get past this error .....
    Pe duminică, 21 august 2022, 12:46:20 EEST, lwvmobile ***@***.***> a scris:  
 
 



I saw that you added the code for LRRP ????


The code is currently able to decode LRRP, but there are some hit or miss samples I have, and also some samples that I don't believe are an actual LRRP or Embedded GPS signalling, but the 3/4 Rate Data seems to convey some form of coordinates. I believe some version of DSDPlus classify this as LOCN.


could we also add an event log???


These is already, in one fashion, an event log of sorts. The Ncurses Terminal window will show call history and also other things of interest. Also, in Linux and the Cygwin Terminal, a user can use combination of the tail command and a selective grep command to pull items of interest out of the log file in realtime, thus making that an event log monitor of sorts as well.

**Comment by lwvmobile** [comment]
> please forget about the error that appears when I try to compile with cygwin the latest version of dsd with LRRP ....... and I don't know how to get past this error .....

I'm going to comb through the code regarding the rtl stuff in the next few days. I've been readying a few updates and you are most likely getting that error because I may have missed placing a conditional ifndef before calling for some of the rtl commands, I'll let you know when I've done that update and tested it.

**Comment by lwvmobile** [comment]
In the future though, if you have errors with compiling, always make a copy of the last bit of terminal output that was leading up to the error. If I can't see the error and read what it says, I can't say for sure what you are missing, or see what I'm doing wrong.

**Comment by romanremus** [comment]
thank you very much.......and I can say that you have a great source code.....if you could add the RC4 encryption in the code, an alert tone when a call occurs, or such a DSD Event log window how is it in dsdplus I can say that you passed dsd plus.....do you have an account where we could contribute money for the code???? because I am very happy with your help
    Pe duminică, 21 august 2022, 12:58:22 EEST, lwvmobile ***@***.***> a scris:  
 
 


In the future though, if you have errors with compiling, always make a copy of the last bit of terminal output that was leading up to the error. If I can't see the error and read what it says, I can't say for sure what you are missing, or see what I'm doing wrong.

**Comment by iz4tow** [comment]
Yes I would like to give too a contribution for your code. Maybe you can publish a donation link on README, I think many OM could enjoy your software and maybe they can contribute too

**Comment by lwvmobile** [comment]
Let me just go on record now and state I will not ask for or accept donations, nor will I accept any sort of money for my work on DSD-FME. It is build on the code of many people of the years! I would never be right to take any sort of money for my contributions (tinkering) to it, when it is built off of the hard work of many other people as well. Not to mention, I would never want to create a situation where a person who did donate had priority in the development or guide or steer what I do because of money. 

**I, nor DSD-FME, will accept any form of monetary compensation or donation.**



> if you could add the RC4 encryption in the code

Again, for the last time, I, for the foreseeable future, will not be adding any forms of advanced or enhanced encryption/decryption. Please refrain from continually asking me.

I am officially closing this conversation now. Please refrain from reopening it. Thank  you.
